src/Servers/Kestrel/test/Interop.FunctionalTests/HttpClientHttp2InteropTests.cs

```diff
@@ -0,0 +1,561 @@
+            await host.StopAsync();
+        }
+
+        private class StreamingContent : HttpContent
+        {
+            private TaskCompletionSource<int> _sendStarted = new TaskCompletionSource<int>(TaskCreationOptions.RunContinuationsAsynchronously);
+            private Func<string, Task> _sendContent;
+            private TaskCompletionSource<int> _sendComplete;
+
+            public StreamingContent()
+            {
+            }
+
+            public async Task SendAsync(string text)
+            {
+                await _sendStarted.Task;
+                await _sendContent(text);
+            }
+
+            public void Complete()
+            {
+                if (_sendComplete == null)
+                {
+                    throw new InvalidOperationException("Sending hasn't started yet.");
+                }
+                _sendComplete.TrySetResult(0);
+            }
+
+            protected override Task SerializeToStreamAsync(Stream stream, TransportContext context)
+            {
+                _sendComplete = new TaskCompletionSource<int>(TaskCreationOptions.RunContinuationsAsynchronously);
+                _sendContent = async text =>
+                {
+                    try
+                    {
+                        var bytes = Encoding.UTF8.GetBytes(text);
+                        await stream.WriteAsync(bytes);
+                        await stream.FlushAsync();
+                    }
+                    catch (Exception ex)
+                    {
+                        _sendComplete.TrySetException(ex);
+                        throw;
+                    }
+                };
+                _sendStarted.SetResult(0);
+                return _sendComplete.Task;
+            }
+
+            protected override bool TryComputeLength(out long length)
+            {
+                length = 0;
+                return false;
+            }
+        }
+
+        private static HttpClient CreateClient()
+        {
+            var handler = new HttpClientHandler();
+            handler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+            var client = new HttpClient(handler);
+            client.DefaultRequestVersion = HttpVersion.Version20;
+            return client;
+        }
+
+        private static HttpRequestMessage CreateRequestMessage(HttpMethod method, string url, HttpContent content)
+        {
+            return new HttpRequestMessage(method, url)
+            {
+                Version = HttpVersion.Version20,
+                Content = content,
+            };
+        }
+
+        private static void ConfigureKestrel(IWebHostBuilder webHostBuilder, string scheme)
+        {
+            webHostBuilder.UseKestrel(options =>
+            {
+                options.Listen(IPAddress.Loopback, 0, listenOptions =>
+                {
+                    listenOptions.Protocols = HttpProtocols.Http2;
```

Comment:
If we're already forcing HTTP/2 on the client, it might be more better/more realistic to keep Kestrel in the default mode.

Reply:
Kestrel can't be in its default most for h2c testing.